Change the FID field to ID

ryanfranz opened the pull request:
Kevin,
I fixed option1.py to use the id field rather than the fid field.  You are welcome to merge my change in, if you want.
Thanks,
Ryan

ryanfranz committed:
Change the FID field to ID

kevinbentley commented:
Thanks, I'll integrate it. Sorry, I haven't got to it yet, I'm in crunch
mode for a demo next week right now. I'll merge that into my Option1d code
as well.

On Thu, Mar 21, 2019 at 3:09 PM ryanfranz <notifications@github.com> wrote:

> Kevin,
> I fixed option1.py to use the id field rather than the fid field. You are
> welcome to merge my change in, if you want.
> Thanks,
> Ryan
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/Cognitics/GeoCDB/pull/1
> Commit Summary
>
>    - Change the FID field to ID
>
> File Changes
>
>    - *M* Converters/Option1.py
>    <https://github.com/Cognitics/GeoCDB/pull/1/files#diff-0> (2)
>
> Patch Links:
>
>    - https://github.com/Cognitics/GeoCDB/pull/1.patch
>    - https://github.com/Cognitics/GeoCDB/pull/1.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Cognitics/GeoCDB/pull/1>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFXWNJWePx_f8jjUWYf_qeW1QTbAdLTYks5vY_UOgaJpZM4cCYLg>
> .
>

ryanfranz committed:
Fix the FID field to be id
Fix floating point numbers to convert correctly

ryanfranz committed:
Fix crashes for shapefiles without a class-level dbf file

ryanfranz commented:
One more change to fix a crash when a shapefile doesn't have a class-level dbf attribute file (caused a crash on Yemen CDB Navigation data)

ryanfranz committed:
Convert logical fields as text, since true/false can be many values and need a better conversion to be a boolean

ryanfranz committed:
Update README.md

Added converter procedure

BernardLeclerc commented on Converters/Option1d.py:
```diff
@@ -281,13 +287,13 @@ def convertShapeFile(shpFilename, cdbInputDir, cdbOutputDir):
     #convertSHP(sqliteCon,shpFilename,outputGeoPackageFile, fClassRecords, True)
     sqliteCon = sqlite3.connect(outputGeoPackageFile)
-    if(createExtendedAttributesTable(sqliteCon,shpFilename)):
+    if(0 and createExtendedAttributesTable(sqliteCon,shpFilename)):
```
Ryan, could you please explain why you want to disable the above `if` statement ?

ryanfranz replied:
This code crashes on Yemen (Tiles\N12\E044\201_RoadNetwork\L00\U0\N12E044_D201_S002_T003_L00_U0_R0.shp).  It is trying to get the CNAM attribute, but from an extended attribute DBF file where one doesn't exist.  Since it wasn't converting correctly anyway, I disabled it.

ryanfranz committed:
Merge pull request #1 from Cognitics/master

Bug fixes for related tables

ryanfranz committed:
Cleanup Option3 and Option4 to work in a similar manner to Option1d (input dir and output dir)
Fixed the setting of class-level attributes, fixed boolean and float fields
Various cleanups to make these files almost identical

ryanfranz committed:
Fix the building of the file lists

ryanfranz committed:
Fix the name of the table in the GeoPackage

ryanfranz committed:
Modified to store the dataset, component selectors, lod, row and column as integers to facilitate faster SQL queries

ryanfranz committed:
Changed to not set the table attributes unless some features exist, so that an empyt class-level attribute file doesn't prevent attributes from being created

ryanfranz committed:
Close the GeoPackage file, so that the spatial indices and extents are populated into the table

ryanfranz committed:
Add creation of a search index for file parts that are not in the table name
